distributed/client.py

```diff
@@ -3625,6 +3635,9 @@ def track_future(self, future):
             else:
                 self.queue.put_nowait(future)
             self._notify()
+        if all([self.with_results, self.raise_errors,
+                future.status == 'error']):
```

Comment:
I have a slight style preference for two `and`s rather than one `all`.  Going over 80 characters is fine.